test/dataframe.jl

```diff
@@ -325,25 +325,28 @@ module TestDataFrame
     df = DataFrame(a=Union{Int, Missing}[1, 2, 3], b=Union{Float64, Missing}[3.0, 4.0, 5.0])
     @test deleterows!(df, [2, 3]) === df
     @test df == DataFrame(a=[1], b=[3.0])
-
-    # describe
-    #suppress output and test that describe() does not throw
-    devmissing = Compat.Sys.isunix() ? "/dev/null" : "nul"
-    open(devmissing, "w") do f
-        @test nothing == describe(f, DataFrame(a=[1, 2], b=Any["3", missing]))
-        @test nothing ==
-              describe(f, DataFrame(a=Union{Int, Missing}[1, 2],
-                                    b=["3", missing]))
-        @test nothing ==
-              describe(f, DataFrame(a=CategoricalArray([1, 2]),
-                                    b=CategoricalArray(["3", missing])))
-        @test nothing == describe(f, [1, 2, 3])
-        @test nothing == describe(f, [1, 2, 3])
-        @test nothing == describe(f, CategoricalArray([1, 2, 3]))
-        @test nothing == describe(f, Any["1", "2", missing])
-        @test nothing == describe(f, ["1", "2", missing])
-        @test nothing == describe(f, CategoricalArray(["1", "2", missing]))
-    end
+    
+    # Test describe
+    struct MySuperLongNameForAStructIsThisToolong end
```

Comment:
Actually it's not very useful here to have a type with a long name, as the printing is done using the standard data frame printing system. Better use e.g. `Date`, and if you want to test printing with long types, add a test in show.jl which wouldn't use `describe` at all.

Also, please add a `CategoricalArray` column as that's a common type.

Reply:
I added a try-catch for `get_stats` when working with `Any` and `Union{Any, Missing}`. This way the user can define a custom type and all the expected overloadings of functions will work. I am not sure of the performance, however. I assume that `try catch` is O(1) since it doesn't take very long to see if there's an error. But I don't know how the compiler works.